app/javascript/lib/action.js

```diff
@@ -64,7 +66,7 @@ export default {
   initialize: makeAction("initialize", ["players", "soloMode", "variant", "baseGame"]),
   blockCanal: makeAction("blockCanal", []),
   bondPurchase: makeAction("bondPurchase", ["nation", "player", "cost", "tradeInValue"]),
-  buildFactory: makeAction("buildFactory", ["province"]),
+  buildFactory: makeAction("buildFactory", ["province", "player", "nationCosts", "playerCosts"]),
```

Comment:
I'm wondering if we can get away with omitting `nationCosts` and `playerCosts`, too. Here's my thinking:
- If the nation can afford the factory, the player MAY NOT supplement with player cash.
- If the nation cannot afford the factory, the player supplements with player cash.
- If both nation and player cannot afford the factory then this would be an invalid action.

The above would be mistaken if the player gets to _choose_ to supplement with their cash even though a nation can afford to build a factory. Is this ever possible? What do you think?

Reply:
Yeah we dont really need them other than for display purposes, and this was the quickest way to pass the data along so that the frontend didn't have to touch any game logic

Up to you, my only intention was to avoid frontend having to pick up the usecases and calculate the costs. When it comes back, the costs are re-calculated to avoid any malicious stuff anyway